janus.c

```diff
@@ -203,7 +206,14 @@ void janus_handle_signal(int signum)
 			JANUS_PRINT("Ok, leaving immediately...\n");
 			break;
 	}
+
+	/* Wake the main loop */
+	janus_mutex_lock(&stop_mutex);
```

Comment:
Indeed, the main loop (which is run by the first/main thread of the process) receives the signal and jumps to the signal handler (even though it's unlikely to come just as the main loop takes the mutex). That's why the signal interrupts the usleep() there (I think g_usleep() is in a loop that sleeps again for the remaining time if its interrupted).